shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -261,9 +271,17 @@ - (void)applicationBecameActive:(NSNotification*)notification {
 }
 
 - (void)applicationWillResignActive:(NSNotification*)notification {
+  [_lifecycleChannel.get() sendMessage:@"AppLifecycleState.inactive"];
+}
+
+- (void)applicationDidEnterBackground:(NSNotification*)notification {
   [_lifecycleChannel.get() sendMessage:@"AppLifecycleState.paused"];
 }
 
+- (void)applicationWillEnterForeground:(NSNotification*)notification {
+  [_lifecycleChannel.get() sendMessage:@"AppLifecycleState.inactive"];
```

Comment:
You are sending the "inactive" message when the application is entering the foreground. I am pretty sure you meant "resumed".